### PR TITLE
Test scenarios for Orchestrator

### PR DESCRIPTION
**Pull request by IrinaComindaru**

### CODE REVIEW

File: tests/features/orchestrator.feature [hunk 1]
```diff
@@ -0,0 +1,65 @@
+Background:
+  Given a valid 'flow' is an object with three valid entries called 'name', 'id' and 'execute'
+
+ Scenario: Orchestrator is initiated with configuration with one valid flow - flow is executed
+  Given Workspace service including service A and service B
+  And   Orchestrator including in its configuration a valid flow between service A and B
+  When  the Workspace starts and Orchestrator is initiated
```
Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]:
```suggestion
  When Orchestrator is initiated
```

File: tests/features/orchestrator.feature [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		 |id      | string   |
+		 |execute | function |
+  When  the Workspace starts and Orchestrator is initiated
+  Then  a relevant error is returned
+
+Scenario: Orchestrator is initiated with a configuration with no flow inside - an error is returned
+    Given Workspace service including service A and service B
+	And   Orchestrator with an empty configuration (no flow inside)
+    When  the Workspace starts and Orchestrator is initiated
+	Then  a relevant error is returned
```
Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]: No need for error in this case, it will just do nothing

File: tests/features/orchestrator.feature [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+Scenario: Orchestrator is initiated with a configuration with empty flow - an error is returned
+    Given Workspace service including service A and service B
+	And   Orchestrator including in its configuration an empty flow
```
Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]: No need for error in this case, it will just do nothing

Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]:
```suggestion
	And   Orchestrator including a valid configuration with an empty list of flows.
```

File: tests/features/orchestrator.feature [hunk 4]
```diff
@@ -0,0 +1,65 @@
+Background:
+  Given a valid 'flow' is an object with three valid entries called 'name', 'id' and 'execute'
```
Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]: Define here what is a valid configuration

File: tests/features/orchestrator.feature [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+         |property| type     |
+		 |name    | string   |
+		 |id      | string   |
+		 |execute | function |
+  When  the Workspace starts and Orchestrator is initiated
+  Then  a relevant error is returned
+
+Scenario: Orchestrator is initiated with a configuration with no flow inside - an error is returned
+    Given Workspace service including service A and service B
+	And   Orchestrator with an empty configuration (no flow inside)
```
Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]:
```suggestion
	And   Orchestrator with an invalid configuration (no flow inside)
```

File: tests/features/orchestrator.feature [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	Then  a relevant error is returned
+
+Scenario: Initiating Orchestrator with empty token or token not a string returns error
+    Given Workspace service including service A and service B
+    And   Orchestrator including in its configuration a valid flow between service A and B
+    And   Orchestrator has no token or a token which is not string
+    When  the Workspace starts and Orchestrator is initiated
+    Then  a relevant error is returned
+
+Scenario: Execute a flow with a function that returns error
```
Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]: Not relevant, the following one is enough

File: tests/features/orchestrator.feature [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        |id      | string   |
+        |execute | function |
+  And   a valid 'configuration' is an object with a unique property called 'flows'
+  And   'flows' is an array of 'flow'
+
+ Scenario: Orchestrator is initiated with configuration with one valid flow - flow is executed
+  Given Workspace service including service A and service B
+  And   Orchestrator including in its configuration a valid flow between service A and B
+  When  Orchestrator is initiated
+  Then  the flow between service A and B is executed
```
Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]: Seems redundant to say `the flow is executed` and `promise is resolved`

File: tests/features/orchestrator.feature [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    When  Orchestrator is initiated
+	Then  the Promise is resolved
+
+ Scenario: The function of one of the flows returns error
+   Given Workspace service including service A and service B
+   And   Orchestrator including in its configuration few  valid flows between service A and B
+   And   One of the flows has an execute function which will return error when executed
+   When  Orchestrator is initiated
+   And   the flows between service A and B are executed
+   And   a notification about the error in execute function is returned
```
Comment (stephanebenayoun) [inline tests/features/orchestrator.feature]: Add a case where a flow try to use a service that doesn't exist in workspace

File: tests/features/orchestrator.feature [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
```
Comment (MikeSquall) [inline tests/features/orchestrator.feature]:
```suggestion
   Then   a notification about the error in execute function is returned
```

**Review by MikeSquall — APPROVED** [review]

**Pull request merged**